File: source/internal/server/api-types/video-bucket.go
package apitypes

// VideoBucketResponse represents the structure of the response for the latest videos.
type VideoBucketResponse struct {
	VideoIDs          []string `json:"videoIds"`
	TotalVideos       int      `json:"totalVideos"`
	CurrentBucket     int      `json:"currentBucket"`
	BucketContentSize int      `json:"bucketContentSize"`
	TotalBuckets      int      `json:"totalBuckets"`
}

File: web/src/ova-angular-sdk/rest-api/api-types/video-bucket.ts
export interface VideoBucketContainer {
  videoIds: string[]; // Array of video IDs
  totalVideos: number; // Total number of videos cached
  currentBucket: number; // The current bucket requested
  bucketContentSize: number; // Size of each bucket (fixed to 20)
  totalBuckets: number; // Total number of buckets
}
